### DISCUSSION TITLE
skip_dir help

### DISCUSSION THREAD

**Discussion by speedster-pta** [abraunegg/onedrive#1468] [Q&A] [answered]
I get an error when running onedrive with the following skip lines:
```
skip_file = ".*|~*" 
skip_dir = "Finances|Transfer"
```
```
Error:
Unknown key in config file: skip_dir
Configuration file has errors - please check your configuration
```
What am I doing wrong?

**Comment by abraunegg** [comment, marked as answer]
@speedster-pta 
> `Unknown key in config file: skip_dir`

If the application you are running does not understand 'skip_dir' in the configuration file you are running an unsupported client version.

You must upgrade your client.

**Comment by speedster-pta** [comment]
I'm running it on an Ubuntu server. What would be the best way to do this?
Do i simply reinstall the package it or is there a better way to go about
it? It doesn't look like apt update works for this

Thank you for a great application!

**Reply by abraunegg** [reply]
@speedster-pta 
Please read the documentation: https://github.com/abraunegg/onedrive/blob/master/docs/INSTALL.md

**Comment by speedster-pta** [comment]
I perused those pages earlier in great detail, and have again now, but
still cannot find any reference to upgrading the package. Could you perhaps
point me in the right direction?

**Reply by abraunegg** [reply]
@speedster-pta 
There is no 'upgrade' process. You have 2 choices:
* Install from a package that is available for your distribution
  * **Note:** Only Alpine Linux, Arch Linux, OpenSuse, Solus keep their packages in-line with code releases. All other packages are out of date and not supported. This is especially important for Ubuntu and Ubuntu based platforms, including the Ubuntu PPA. 
* Build from source. This does not change your data or change your configuration but `make install` you are installing the new binary, which is 'upgrading' your old one.

**Additional note:** Based on your configuration file entries, you are also using the 'skilion' codebase. You must first uninstall & ensure that that client is 100% removed from your platform before hand. This is because that client does not install / uninstall correctly to begin with, thus, must be properly & fully removed manually to ensure the actual binary 'onedrive' is not left behind on your system anywhere.

**Comment by speedster-pta** [comment]
Thank you kindly
